test/partition-test/puml.cpp

```diff
@@ -0,0 +1,451 @@
+	}
+}
+
+int main(int argc, char* argv[])
+{
+	int p;
+	MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &p);
+
+	MPI_Comm comm = MPI_COMM_WORLD;
+	int rank, size;
+	MPI_Comm_rank(comm, &rank);
+	MPI_Comm_size(comm, &size);
+
+	if (argc == 1) {
+		logWarning(rank) << "No settings given. They can be set as command line arguments. "
+		    "An argument can have two forms, either a key-value pair in the form \"key=value\" (note the '=', and no spaces around it), "
+			"or it can be a file name which contains a JSON file (note in particular: you may mix files and key-value pairs, and you may supply multiple files). "
+			"Later arguments override earlier arguments. If a file contains an array of configurations, each configuration is tested."
+			"Any following argument is then applied to all configurations. If again an array of configurations is given, the cross product between the two arrays is used.";
+	}
+
+	auto settingsList = argparse(argc, argv);
+
+	
+	logInfo(rank) << "PUML Partition Tester --- Starting";
+	logInfo(rank) << settingsList.size() << "tests";
+
+	PUML::TETPUML* puml = nullptr;
+
+	std::string lastmesh = "";
+	std::string lastmatmesh = "";
+	std::string lastmat = "";
+
+	for (int i = 0; i < settingsList.size(); ++i) {
+		auto settings = settingsList[i];
+		logInfo(rank) << "Test" << (i+1) << "of" << settingsList.size();
+		logInfo(rank) << "Using configuration: " << settings;
+		if (settings.is_null()) {
+			logInfo(rank) << "Aborting...";
+			break;
+		}
+
+		if (puml == nullptr || readsetting<std::string>(settings, "meshfile") != lastmesh) {
+			delete puml;
+			puml = new PUML::TETPUML();
+			std::string infile = readsetting<std::string>(settings, "meshfile");
+			logInfo(rank) << "Reading mesh" << infile;
+			puml->open((infile + ":/connect").c_str(), (infile + ":/geometry").c_str());
+
+			logInfo(rank) << "Reading other data";
+			puml->addData((infile + ":/group").c_str(), PUML::CELL);
+			puml->addData((infile + ":/boundary").c_str(), PUML::CELL);
+
+			logInfo(rank) << "Generating mesh information";
+			puml->generateMesh();
+
+			lastmesh = readsetting<std::string>(settings, "meshfile");
+		}
+
+		std::string partitionerName = readsetting<std::string>(settings, "name");
+		auto partitionInvoker = PUML::TETPartition::get_partitioner(partitionerName);
+		if (partitionInvoker == nullptr) {
+			logInfo(rank) << "Partitioner name not recognized. Aborting...";
+			MPI_Finalize();
+			return 1;
+		}
+
+		logInfo(rank) << "Creating partitions using partitioner" << partitionerName;
+		std::vector<int> partition(puml->numOriginalCells());
+		PUML::TETPartitionGraph graph(*puml);
+
+		std::string vertexweightsetting = readsetting<std::string>(settings, "vw");
+		std::string edgeweightsetting = readsetting<std::string>(settings, "ew");
+		std::string nodeweightsetting = readsetting<std::string>(settings, "nw");
+
+		int * vertexweights = nullptr;
+		int * edgeweights = nullptr;
+		double * nodeweights = nullptr;
```

Comment:
why aren't they std::vectors?

Reply:
That was the code back in 2019—as said above, I'd leave the tests for now mostly alone and postpone re-working them properly.